yt/geometry/grid_geometry_handler.py

```diff
@@ -475,7 +475,16 @@ def _mesh_sampling_particle_field(field, data):
 
             i, j, k = np.floor((pos - data.LeftEdge) / data.dds).astype("int64").T
 
-            return field_values[i, j, k]
+            # Make sure all particles are within the current grid, otherwise return nan
+            maxi, maxj, maxk = field_values.shape
+
+            mask = (i < maxi) & (j < maxj) & (k < maxk)
+            mask &= (i >= 0) & (j >= 0) & (k >= 0)
+
+            result = np.full_like(i, np.nan, dtype="float64")
```

Comment:
I'm confused. Why is is `full_like(i)` ?

Reply:
I could instead use 
```suggestion
            result = np.full_like(field_value, np.nan, dtype="float64")
```

Reply:
Please do, it's a bit clearer :)